File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    let honor = [];
    for(let i = 0; i < score.length; i++){
        honor.push(score[i]);
        honor.sort((a, b) => b- a);
        answer.push(Math.min(...honor.slice(0, k)));
    }
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    temp = []
    for val in score:
        
        if len(temp) < k:
            temp.append(val)
            temp.sort(reverse=True)
            answer.append(temp[-1])
        else:
            temp.append(val)
            temp.sort(reverse=True)
            answer.append(temp[k-1])
    return answer